OpenSim/Simulation/GeodesicWrap/GeodesicPath.h

```diff
@@ -0,0 +1,85 @@
+#ifndef OPENSIM_GEODESIC_PATH_H_
+#define OPENSIM_GEODESIC_PATH_H_
+
+// INCLUDE
+
+#include "OpenSim/Simulation/GeodesicWrap/GeodesicCurveState.h"
+#include "GeodesicWrapObject.h"
+#include "GeodesicPathSolver.h"
+
+namespace OpenSim {
+
+// Notes:
+// - rename Object to Surface?
+// - GeodesicWrapObject holds pointer to frame, how?
+// - ImplicitSurface holds the local state, how can it cache it?
+// - what is the workflow?
+//     - create surface component, add to model, connect to frame.
+//     - create wrapping path, add to model, connect each surface.
+//     - what are those connections? Sockets? Parameters? shared-pointers?
+
+class AbstractGeometryPath;
+
+// Concrete implementation of the AbstractGeometryPath.
+//
+// Contains the solver for computing the entire wrapping path.
+//
+// Comparable to GeometryPath?
+//
+// Does the following:
+// - Caches the result from the solver.
+// - Holds set of WrapOstacles
+// - Implements AbstractGeometryPath
+class OSIMSIMULATION_API GeodesicPath : public AbstractGeometryPath {
+OpenSim_DECLARE_CONCRETE_OBJECT(GeodesicPath, ModelComponent);
+public:
+    //==============================================================================
+    // PROPERTIES
+    //==============================================================================
+    OpenSim_DECLARE_PROPERTY(some_property, std::string,
+        "some property");
+
+    //--------------------------------------------------------------------------
+    // CONSTRUCTION
+    //--------------------------------------------------------------------------
+public:
+    GeodesicPath() = default;
+    ~GeodesicPath() = default;
+
+//=============================================================================
+// METHODS
+//=============================================================================
+
+    //--------------------------------------------------------------------------
+    // ABSTRACTGEOMETRYPATH IMPLEMENTATION
+    //--------------------------------------------------------------------------
+    double getLength(const SimTK::State& s) const override;
+    double getLengtheningSpeed(const SimTK::State& s) const override;
+    void addInEquivalentForces(const SimTK::State& state,
+    const double& tension,
+    SimTK::Vector_<SimTK::SpatialVec>& bodyForces,
+    SimTK::Vector& mobilityForces) const override;
+    double computeMomentArm(const SimTK::State& s,
+    const Coordinate& aCoord) const override;
+    bool isVisualPath() const override;
+
+private:
+
+    // Path start and end points.
+    SimTK::Vec3 _pStart;
+    SimTK::Vec3 _pEnd;
+
+    // List of wrap objects.
+    std::vector<GeodesicWrapObject> _wrapObstacles;
+
+    // Buffer for holding the shooter results.
+    std::vector<GeodesicCurve> _surfaceShooterResults;
```

Comment:
As discussed, we could store this using `SimTK::DiscreteVariable`, which would allow us to store the geodesic between integration steps and update after each geodesic solver convergence.

One thing to consider: how to populate this variable when loading a model from XML? We'd need to make sure that the default `GeodesicCurve` values represent a converged value for the path in the model's default kinematic configuration.

Reply:
I must say, I'm completely unsure. We could start with recomputing the path every single time. Im currently still not sure who is actually owning the data and what the data is :)
My naive guess would be to store in `GeometryPath` a `std::vector<(startPosition, startVelocity, length)>`? That should be enough to encode any geodesic.

Reply:
In my branch, I've temporarily added a `default_geodesics` property which could be used to initialized paths when loaded from file. But I think your suggestion is better, since storing the whole geodesic is probably not necessary. Also, the start and end (after shooting) positions for each geodesic could be used to initialize the straight line segments for each `GeodesicPathSegment`. 

The simpler we can make the path initialization here, the easier it will be to build tools for path initialization design. We just need to find a way to create initial conditions that generate a valid geodesics (including unit velocity normalization, etc). After that, it's just a matter of connecting the dots.